installation fails with codesign command not found

I executed the following command on my Mac m1

But the build fails with following error
"
/nix/store/nqld6s67abqzkdy90rkgz5bqbpsfcy8s-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CoreText.h:61:10: note: 'CTGetCoreTextVersion' has been explicitly marked deprecated here
uint32_t CTGetCoreTextVersion( void ) CT_DEPRECATED("Use -[NSProcessInfo operatingSystemVersion]", macos(10.5, 11.0), ios(3.2, 14.0), watchos(2.0, 7.0), tvos(9.0, 14.0));
         ^
1 warning generated.
  CCLD     temacs
codesign -s - -f temacs.tmp
/nix/store/igfm90d0b7nij5xi1hgr63sdk4fkzjng-bash-4.4-p23/bin/bash: codesign: command not found
make[1]: *** [Makefile:644: temacs] Error 127
make[1]: Leaving directory '/private/tmp/nix-build-emacs-osx-20210925.0.drv-0/source/src'
make: *** [Makefile:443: src] Error 2
builder for '/nix/store/wg16g72vkicijfwqqcn5g6gpaw1bb2lw-emacs-osx-20210925.0.drv' failed with exit code 2
error: build of '/nix/store/wg16g72vkicijfwqqcn5g6gpaw1bb2lw-emacs-osx-20210925.0.drv' failed
"

--- a/emacs-overlay.nix
+++ b/emacs-overlay.nix
@@ -34,18 +34,24 @@ let
         }))
     ];
 
-  emacsOsx = mkGitEmacs "emacs-osx" ./emacs-source/emacs-master.json [ ] { };
+  emacsOsx = mkGitEmacs "emacs-osx" ./emacs-source/emacs-master.json [
+    ./patches/codesign.patch
+  ] { };
 
-  emacsOsxNative = mkGitEmacs "emacs-osx" ./emacs-source/emacs-master.json [ ] {
+  emacsOsxNative = mkGitEmacs "emacs-osx" ./emacs-source/emacs-master.json [
+    ./patches/codesign.patch
+  ] {
     nativeComp = true;
   };
 
   emacsOsxTile = mkGitEmacs "emacs-osx" ./emacs-source/emacs-master.json [
+    ./patches/codesign.patch
     ./patches/no-titlebar.patch
     ./patches/fix-window-role-yabai.patch
   ] { };
 
   emacsOsxNativeTile = mkGitEmacs "emacs-osx" ./emacs-source/emacs-master.json [
+    ./patches/codesign.patch
     ./patches/no-titlebar.patch
     ./patches/fix-window-role-yabai.patch
   ] { nativeComp = true; };

--- a/patches/codesign.patch
+++ b/patches/codesign.patch
@@ -1,0 +1,19 @@
+--- a/src/Makefile.in
++++ b/src/Makefile.in
+@@ -698,7 +698,7 @@
+ ifeq ($(HAVE_PDUMPER),yes)
+ 	$(AM_V_at)$(MAKE_PDUMPER_FINGERPRINT) $@.tmp
+ ifeq ($(DO_CODESIGN),yes)
+-	codesign -s - -f $@.tmp
++	/usr/bin/codesign -s - -f $@.tmp
+ endif
+ endif
+ 	$(AM_V_at)mv $@.tmp $@
+@@ -901,7 +901,7 @@
+ 	@: In the pdumper case, make compile-first after the dump
+ 	cp -f temacs$(EXEEXT) bootstrap-emacs$(EXEEXT)
+ ifeq ($(DO_CODESIGN),yes)
+-	codesign -s - -f bootstrap-emacs$(EXEEXT)
++	/usr/bin/codesign -s - -f bootstrap-emacs$(EXEEXT)
+ endif
+ endif

